k8s/upgrades/0.8.0-0.8.1/cstor_volume_upgrade.sh

```diff
@@ -97,49 +101,44 @@ c_rs=$(kubectl get rs -o name --namespace $ns | grep $c_dep | cut -d '/' -f 2)
 # previous step                                                #  
 ################################################################
 
-sed "s/@sc_name/$sc_name/g" cstor-target-patch.tpl.json | sed -u "s/@sc_resource_version/$sc_res_ver/g" > cstor-target-patch.json
-sed "s/@sc_name/$sc_name/g" cstor-target-svc-patch.tpl.json | sed -u "s/@sc_resource_version/$sc_res_ver/g" > cstor-target-svc-patch.json
-sed "s/@sc_name/$sc_name/g" cstor-volume-patch.tpl.json | sed -u "s/@sc_resource_version/$sc_res_ver/g" > cstor-volume-patch.json
-sed "s/@sc_name/$sc_name/g" cstor-volume-replica-patch.tpl.json | sed -u "s/@sc_resource_version/$sc_res_ver/g" > cstor-volume-replica-patch.json
+sed "s/@sc_name/$sc_name/g" cstor-target-patch.tpl.json | sed -u "s/@sc_resource_version/$sc_res_ver/g" | sed -u "s/@target_version/$Target_upgrade_version/g" > cstor-target-patch.json
+sed "s/@sc_name/$sc_name/g" cstor-target-svc-patch.tpl.json | sed -u "s/@sc_resource_version/$sc_res_ver/g" | sed -u "s/@target_version/$Target_upgrade_version/g" > cstor-target-svc-patch.json
+sed "s/@sc_name/$sc_name/g" cstor-volume-patch.tpl.json | sed -u "s/@sc_resource_version/$sc_res_ver/g" | sed -u "s/@target_version/$Target_upgrade_version/g" > cstor-volume-patch.json
+sed "s/@sc_name/$sc_name/g" cstor-volume-replica-patch.tpl.json | sed -u "s/@sc_resource_version/$sc_res_ver/g" | sed -u "s/@target_version/$Target_upgrade_version/g" > cstor-volume-replica-patch.json
 
 #################################################################################
 # STEP: Patch OpenEBS volume deployments (cstor-target, cstor-svc)              #  
 #################################################################################
 
 
 # #### PATCH TARGET DEPLOYMENT ####
-echo "Upgrading Target Deployment to 0.8.1"
+echo "Upgrading Target Deployment to $Target_upgrade_version"
 
 # Setting deployment startegy to recreate
 setDeploymentRecreateStrategy $ns $c_dep
 
 kubectl patch deployment  --namespace $ns $c_dep -p "$(cat cstor-target-patch.json)" 
```

Comment:
I guess each patch should have its own check for version !!
e.g. if a resource say CVR is getting patched, then its version can be checked. 
- If version matches target version then continue to next resource's patching 
- If the version matches expected current version then upgrade

Reply:
Above needs to be done for every resource !!!
This is more about validation handling & whether to skip or continue !!
Think about the cases where upgrade failed first time, then the user retries. The current code may not allow if some of the resources were upgraded successfully while some others failed.

Reply:
cool tried to implement the same